pkg/rtc/participant.go

```diff
@@ -664,6 +674,68 @@ func (p *ParticipantImpl) GetAudioLevel() (level uint8, active bool) {
 	return
 }
 
+func (p *ParticipantImpl) GetConnectionQuality() livekit.ConnectionQuality {
+	// avg loss across all tracks, weigh published the same as subscribed
+	var pubLoss, subLoss uint32
+	var reducedQualityPub bool
+	var reducedQualitySub bool
+	p.lock.RLock()
+	defer p.lock.RUnlock()
+	for _, pubTrack := range p.publishedTracks {
+		if pubTrack.IsMuted() {
+			continue
+		}
+		pubLoss += pubTrack.PublishLossPercentage()
+		publishing, registered := pubTrack.NumUpTracks()
+		if registered > 0 && publishing != registered {
+			reducedQualityPub = true
+		}
+	}
+	numTracks := uint32(len(p.publishedTracks))
+	if numTracks > 0 {
+		pubLoss /= numTracks
```

Comment:
Would be good to change this at some point to take aggregate packet count and do percentages. With DTX, this could skew the calculation.

Reply:
that's a great point!